warden/x/warden/keeper/msg_server_new_sign_request.go

```diff
@@ -66,3 +62,26 @@ func (k msgServer) NewSignRequest(ctx context.Context, msg *types.MsgNewSignRequ
 
 	return &types.MsgNewSignRequestResponse{Id: id}, nil
 }
+
+func chargeKeychainFee(k *msgServer, ctx context.Context, requestedFee string, keychainAccAddress sdk.AccAddress, keychainFees sdk.Coins, creator string) error {
+	feeCoin, err := sdk.ParseCoinNormalized(requestedFee)
+	if err != nil {
+		return fmt.Errorf("invalid format fee passed: %w", err)
+	}
+
+	found, feeInRequestedCoin := keychainFees.Find(feeCoin.Denom)
```

Comment:
technically a keychain can specify a fee composed of different denoms (e.g. `1uward` AND `1usdc`) so this scenario wouldn't work here, were only one denom is expected to be specified in `requestedFee`

Reply:
Thank you, fixed